cmd/mapt/cmd/aws/hosts/fedora.go

```diff
@@ -50,13 +52,28 @@ func getFedoraCreate() *cobra.Command {
 				viper.GetString(params.ConnectionDetailsOutput),
 				viper.GetStringMapString(params.Tags))
 
+			// Get the VM instance type name based on cpus, memory and arch
+			var ec2Types []string
+			if viper.GetInt32(params.CPUs) > 0 && viper.GetInt32(params.Memory) > 0 {
+				var arch resources.Arch = resources.Amd64
+				if viper.GetString(params.LinuxArch) == "arm64" {
+					arch = resources.Arm64
+				}
+				ec2Types, _ = resources.GetAwsMachineTypes(viper.GetInt32(params.CPUs), viper.GetInt32(params.Memory), arch)
```

Comment:
This logic needs to be moved to the pkg; inside the actions...

the rational behind... is at the moment the interface to interact with mapt is through cmd,  but in case wanna support the k8s controller model it will directly use the logic on actions as so we need to accept the requested features there and search for the type of machine there

Reply:
so iiuc the required `cpus` and `memories` should be part of the `Request` struct on actions and these will be what passed from the cmd layer and pkg will figure out the instance types based on the cpus,memory and arch

Reply:
yeah, probably the search module will have an struct with those values you use to query (CPU, Ram, Nested Virtualization, GPU and arch) that struct will be part of the request struct

Reply:
key part any logic should be encapsulated within actions, then cmd or controllers just fill the request structs there

Reply:
added `CPUs` and `Memory` fields to the `Request` struct itself, and functions to fetch the instance types

but as you suggest encapsulating these in a new  request structs would be better, then we can have methods on it to fetch the instance types.. e.g named `AwsInstanceTypeRequest` `AzureInstanceTypeRequest`